test/e2e/framework/pod/resource.go

```diff
@@ -701,3 +703,27 @@ func PatchContainerImages(containers []v1.Container) error {
 
 	return nil
 }
+
+// GetKubeletPods retrieves the list of pods on the kubelet.
+func GetKubeletPods(c clientset.Interface, node string) (*v1.PodList, error) {
+	return getKubeletPods(c, node, "pods")
+}
+
+// GetKubeletRunningPods retrieves the list of running pods on the kubelet. The pods
+// includes necessary information (e.g., UID, name, namespace for
+// pods/containers), but do not contain the full spec.
+func GetKubeletRunningPods(c clientset.Interface, node string) (*v1.PodList, error) {
+	return getKubeletPods(c, node, "runningpods")
+}
+
+func getKubeletPods(c clientset.Interface, node, resource string) (*v1.PodList, error) {
+	result := &v1.PodList{}
+	client, err := e2enode.ProxyRequest(c, node, resource, ports.KubeletPort)
```

Comment:
I could see this method being in `framework/node` instead just because it calls `e2enode.ProxyRequest` (simplify the dependency graph) and it essentially stands for "let me see all pods running on this node".
But this is a small nit.
wdyt @andrewsykim @WanLinghao @timothysc ?

Reply:
Do you mean move all those three methods to framework/node? I am OK with that

Reply:
Actually, I just remebered that you created a `framework/kubelet` pkg so that may be even better.
But yes, those 3. 

I originally put the TODO comments to move them onto `framework/pods` just because they have the word "pod" in the name. However, calling the method is more about getting info about the kubelet/node than about managing pods.

Reply:
I think the best solution is moving them to framework/kubelet/kubelet_pod.go, of course, it's a new file. Since currently we have only one file beneath framework/kubelet, and it's more like something `stats` like cpu, memory...
WDYT?